docs/trace/extending-the-sdk/README.md

```diff
@@ -334,6 +339,179 @@ Custom resource detectors can be implemented:
+                {
+                    // Support configuration through Options API.
+                    services.Configure(name, configureExporterOptions);
+                }
+
+                if (configureBatchProcessorOptions != null)
+                {
+                    // Support configuration through Options API.
+                    services.Configure<ExportActivityProcessorOptions>(
+                        name,
+                        o => configureBatchProcessorOptions(o.BatchExportProcessorOptions));
+                }
+
+                // Register custom service as a singleton.
+                services.TryAddSingleton<MyCustomService>();
+            });
+
+            builder.ConfigureBuilder((sp, builder) =>
+            {
+                // Retrieve MyExporterOptions instance using name.
+                var options = sp.GetRequiredService<IOptionsMonitor<MyExporterOptions>>().Get(name);
+
+                // Retrieve MyCustomService singleton.
+                var myCustomService = sp.GetRequiredService<MyCustomService>();
+
+                // Registers MyCustomExporter with a batch processor.
+                builder.AddExporter(
+                    ExportProcessorType.Batch,
+                    new MyCustomExporter(options, myCustomService),
+                    name,
+                    configure: null);
+            });
+
+            // Return builder for call chaining.
+            return builder;
+        }
+    }
+}
+
+namespace MyLibrary
+{
+    // Options class can be bound to IConfiguration or configured by code.
+    public class MyExporterOptions
+    {
+        public Uri? IngestionUri { get; set; }
+    }
+
+    internal sealed class MyCustomExporter : BaseExporter<Activity>
+    {
+        public MyCustomExporter(
+            MyExporterOptions options,
+            MyCustomService myCustomService)
+        {
+            // Implementation not shown.
+        }
+
+        public override ExportResult Export(in Batch<Activity> batch)
+        {
+            // Implementation not shown.
+
+            return ExportResult.Success;
+        }
+    }
+
+    internal sealed class MyCustomService
+    {
+        // Implementation not shown.
+    }
+}
+```
+
+**Note** Use the `OpenTelemetry.Trace` namespace for registration extensions to
+help with discoverability.
+
+**Note** Return the `TracerProviderBuilder` passed in to support call chaining
+of registration methods.
+
+**Note** Use the
+[Configure](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.optionsservicecollectionextensions.configure#microsoft-extensions-dependencyinjection-optionsservicecollectionextensions-configure-1(microsoft-extensions-dependencyinjection-iservicecollection-system-string-system-action((-0))))
+extension to register callbacks to configure options class instances instead of
+invoking callbacks directly.
+
+### IServiceCollection detached registration
```

Comment:
Is "detached registration" a term that is commonly used? Took me a bit to orient myself to what this section was about. Would you say that this scenario would be common for applications not using one of the host builders?

Reply:
I see, maybe little bit of reorganization could help bring some more clarity.

```md
## Guidance for library authors for providing extension methods to configure their custom components

Library authors are encouraged to provide extension methods on the
`TracerProviderBuilder` class for registering custom OpenTelemetry components.
Additionally, library authors may also consider providing extension methods
on `IServiceCollection`. Both of these patterns are described below.

### TracerProviderBuilder extension methods

Targeting `TracerProviderBuilder` will allow users to discover extensions using
IDE helpers such as `IntelliSense` (in Visual Studio or Visual Studio Code)
while configuring the builder(s) in their applications.

...

### IServiceCollection extension methods

In some scenarios, it may also be desirable to offer extension methods on
`IServiceCollection`. This can be the case when an author wishes to simplify
the configuration of numerous custom components. For example, consider the following:

... example of calling three methods here ...

...

The `ConfigureOpenTelemetryTracing` extension method allows this to be simplified.

... IServiceCollection example ...

```

Reply:
I just pushed big changes. Tried to incorporate 1) your feedback from above and 2) feedback from the SIG yesterday. Primarily I included in the `IServiceCollection` example @martinjt's idea to have the extension use options for turning on/off telemetry. Also adjusted the recommendations so each has a more defined use case.

LMK your thoughts!